src/map_extras.cpp

```diff
@@ -76,6 +76,164 @@ void mx_null( map &, const tripoint & )
     debugmsg( "Tried to generate null map extra." );
 }
 
+void mx_roa_crash( map &m, const tripoint &abs_sub )
+{
+    int cx = rng( 6, SEEX * 2 - 7 );
+    int cy = rng( 6, SEEY * 2 - 7 );
+
+    for( int x = 0; x < SEEX * 2; x++ ) {
+        for( int y = 0; y < SEEY * 2; y++ ) {
+            if( m.veh_at( tripoint( x,  y, abs_sub.z ) ) &&
+                m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+            } else {
+                if( x >= cx - dice( 1, 5 ) && x <= cx + dice( 1, 5 ) && y >= cy - dice( 1, 5 ) &&
+                    y <= cy + dice( 1, 5 ) ) {
+                    if( one_in( 7 ) && m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
```

Comment:
The check for the DIGGABLE flag should be done first at the very beginning of the loop body. If it fails, the rest of the loop body is not needed and one can `continue` instead.